packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -219,6 +220,14 @@ class PaginatedDataTable extends StatefulWidget {
   /// {@macro flutter.widgets.scrollable.dragStartBehavior}
   final DragStartBehavior dragStartBehavior;
 
+  /// {@template flutter.material.dataTable.checkboxHorizontalMargin}
+  /// Horizontal margin between the edges of the table and the checkbox as well as
+  /// the content in the first data column, when checkbox is displayed.
+  /// {@endtemplate}
+  ///
+  /// This value defaults to 24.0 to adhere to the Material Design specifications.
```

Comment:
I think this can be updated to (no need for template):
```
Horizontal margin around the checkbox, if it is displayed.

If null, then [horizontalMargin] is used as the margin between the edge
of the table and the checkbox, as well as the margin between the checkbox
and the content in the first data column.
```

I think you can update `horizontalMargin` too to make it clear how it is used together with the new property:
```
The horizontal margin between the edges of the table and the content
in the first and last cells of each row.

This value defaults to 24.0 to adhere to the Material Design specifications.

If [checkboxHorizontalMargin] is null, then [horizontalMargin] is also the 
margin between the edge of the table and the checkbox, as well as the
margin between the checkbox and the content in the first data column.
```